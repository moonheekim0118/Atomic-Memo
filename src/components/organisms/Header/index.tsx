import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import Span from '../../atoms/Span';
import Button from '../../atoms/Button';

interface Props {
  onToggle?: (e: React.MouseEvent<HTMLSpanElement>) => void;
  pageName: string;
}
const Header = ({ onToggle, pageName }: Props) => {
  return (
    <StyledHeader>
      <Button onClick={onToggle} color="#3399ff">
        Menu
      </Button>
      <Span>{pageName}</Span>
      {pageName === 'Home' ? (
        <Link to="/trash">
          <Span>Trash</Span>
        </Link>
      ) : (
        <Link to="/">
          <Span>Home</Span>
        </Link>
      )}
    </StyledHeader>
  );
};

const StyledHeader = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 50px;

  height: 70px;
  border-bottom: 1px solid #f4f4f4;
`;

export default Header;
