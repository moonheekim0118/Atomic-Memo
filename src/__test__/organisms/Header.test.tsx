import 'jsdom-global/register';
import React from 'react';
import Header from '../../components/organisms/Header';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

describe('<Header/>', () => {
  let container = null;
  let mockFn = jest.fn();

  it('renders correctly', () => {
    container = mount(
      <MemoryRouter>
        <Header onToggle={mockFn} pageName="Home" />
      </MemoryRouter>
    );
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
