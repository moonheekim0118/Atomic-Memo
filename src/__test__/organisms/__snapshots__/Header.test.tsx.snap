// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> matches snpashot 1`] = `"<header class=\\"Header__StyledHeader-ooc1x8-0 gXPHBW\\"><button color=\\"#3399ff\\" class=\\"Button__StyledButton-hxb5gx-0 dfmXPz\\">Menu</button><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG normal\\">Home</span><a href=\\"/trash\\"><span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG normal\\">Trash</span></a></header>"`;
